.circleci/config.yml

```diff
@@ -306,44 +312,100 @@ jobs:
           path: _build/dev3/rel/epoch/log
           destination: node3/
 
-  docker_image_build:
+  docker_image_build_local:
     <<: *container_config
     steps:
       - checkout
+      - *merge_checkout
       - *setup_remote_docker
       - *install_docker_client
       - run:
-          name: Build Docker image
+          name: Build and tag Docker image
           command: |
-            ~/bin/docker build .
+            ~/bin/docker build -t ${DOCKERHUB_REPO:?}:local .
 
-  docker_push_tag:
+  docker_image_build_dev:
+    <<: *container_config
+    steps:
+      - checkout
+      - *setup_remote_docker
+      - *install_docker_client
+      - run:
+          name: Build and tag Docker image
+          command: |
+            ~/bin/docker build -t ${DOCKERHUB_REPO:?}:dev ${DOCKERHUB_REPO:?}:local .
+
+  docker_image_build_tag:
     <<: *container_config
     steps:
       - checkout
       - *setup_remote_docker
       - *install_docker_client
       - run:
-          name: Build and push Docker image to DockerHub
+          name: Build and tag Docker image
+          command: |
+            ~/bin/docker build -t ${DOCKERHUB_REPO:?}:${CIRCLE_TAG:?} -t ${DOCKERHUB_REPO:?}:latest ${DOCKERHUB_REPO:?}:local .
+
+  docker_push_tag:
+    <<: *container_config
+    steps:
+      - *setup_remote_docker
+      - *install_docker_client
+      - run:
+          name: Push Docker image to DockerHub
           command: |
             ~/bin/docker login -u $DOCKER_USER -p $DOCKER_PASS
-            ~/bin/docker build -t ${DOCKERHUB_REPO:?}:${CIRCLE_TAG:?} -t ${DOCKERHUB_REPO:?}:latest .
```

Comment:
Last time I checked the built image was not cached/shared between jobs/steps. Can you prove this is working now?

Reply:
Not checked so I expect same error. Tests do not arrive at stage of using tag yet (earlier enoent error).